src/main/java/org/opensearch/ad/ml/EntityColdStarter.java

```diff
@@ -220,6 +220,22 @@ private void coldStart(
     ) {
         logger.debug("Trigger cold start for {}", modelId);
 
+        if (modelState == null || entity == null) {
```

Comment:
If data isn't present currently but will get ingested in the future and we are expecting a long initialization would either of this condition be met until then?

Reply:
This is method invariant and we don't any expect any of them to be true, regardless of whether data is present or not.